napi_from_js.md

```diff
@@ -0,0 +1,367 @@
+# N-API Function APIs 
+
+N-API provides a set of APIs that allow JavaScript code to 
+call back into native code. N-API APIs that support calling back
+into native code take in a callback functions represented by 
+the `napi_callback` type. When the JavaScript VM calls back to
+native code, the `napi_callback` provided is invoked. The APIs 
+documented in this section allow the callback function to do the 
+following:
+- Get information about the context in which the callback was invoked
+- Get the arguments passed into the callback
+- Return a `napi_value` back from the callback
+
+Additionally, N-API provides a set of functions which allow calling 
+JavaScript functions from native code. One can either call a function
+like a regular JavaScript function call, or as a constructor 
+function.
+
+## Functions
+
+### *napi_call_function*
+
+#### Signature
+```C
+napi_status napi_call_function(napi_env env,
+                               napi_value recv,
+                               napi_value func,
+                               int argc,
+                               const napi_value* argv,
+                               napi_value* result)
+```
+
+#### Parameters
+- `[in]  env `: The environment that the API is invoked under
+- `[in]  recv`: The `this` object passed to the called function
+- `[in]  func`: `napi_value` representing the JavaScript function
+to be invoked
+- `[in]  argc`: The count of elements in the `argv` array
+- `[in]  argv`: Array of JavaScript objects as `napi_value` 
+representing the arguments to the function. If there are more arguments
+than the provided argument count, then only the requested number of 
+arguments are copied. If there are fewer arguments provided than 
+claimed, the rest of `argv` is filled with `napi_value` values that 
+represent `undefined`
+- `[out] result`: The JavaScript object representing the return 
+value from the function call
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This method allows you to call a JavaScript function object from your native 
+add-on. This is the primary mechanism of calling back *from* the add-on's
+native code *into* JavaScript.
+
+A sample use case might look as follows. If you had the following JavaScript
+consider the following snippet:
+```js
+function AddTwo(num)
+{
+    return num + 2;
+}
+```
+
+Then, the above function can be invoked from your native add-on using the
```

Comment:
Hmm, I don't know 😄 I'm treating that as a new sentence, and treating the snippet before as a sentence separator. I don't know if it makes sense to put a period at the end of the previous sentence or the snippet that follows it- but I think it reads weird if this particular Then starts with a lower case